### DISCUSSION TITLE
Methods openpty(3C), forkpty(3C) and login_tty(3C)

### DISCUSSION THREAD

**Discussion by fronald** [OpenIndiana/oi-userland#15130] [Ideas]
Anyone knows if this was implemented?

https://www.illumos.org/issues/5386

I'm trying (in my spare time) build and run [VSCodium](https://vscodium.com/) on OpenIndiana. To do it, i've been create some patches on Node.JS native modules that support Linux, macOS and FreeBSD only. Things like this: https://github.com/mongodb-js/kerberos/pull/166

Now, i need make a patch to node-pty (https://github.com/microsoft/node-pty), and i'm not finding these methods (openpty, forkpty and login_tty). Anyone knows anything about these methods?

I found it https://code.illumos.org/c/illumos-gate/+/1343/8/usr/src/uts/common/sys/termios.h, but on searching about termios.h on IPS of OpenIndiana, i have two results: 

```bash
$ pkg search termios.h
INDEX      ACTION VALUE                     PACKAGE
basename   file   usr/include/sys/termios.h pkg:/system/header@0.5.11-2023.0.0.21898
basename   file   usr/include/termios.h     pkg:/system/header@0.5.11-2023.0.0.21898
```
But none have openpty, forkpty or login_tty methods.

**Comment by mtelka** [comment]
The work you found is not integrated into illumos yet.

**Reply by fronald** [reply]
> The work you found is not integrated into illumos yet.

I will try a workaround...
